File: src/shared/utils/coreCall.lua
local MAX_RETRIES = 8

local StarterGui = game:GetService('StarterGui')
local RunService = game:GetService('RunService')

return function(method, ...)
	local result = {}
	for retries = 1, MAX_RETRIES do
		result = {pcall(StarterGui[method], StarterGui, ...)}
		if result[1] == true then
			break
		end
		RunService.Stepped:Wait()
	end
	return unpack(result)
end

File: places/start/src/shared/utils/coreCall/coreCall.ts
import { RunService, StarterGui } from "@rbxts/services";

const MAX_RETRIES = 8;

export function coreCall(method: keyof StarterGui, ...args: unknown[]) {
	const result = [];
	for (let i = 0; i < MAX_RETRIES; i++) {
		const pcallResult = pcall(() => {
			(StarterGui[method] as (this: StarterGui, ...args: unknown[]) => void)(...args);
		});
		result[0] = pcallResult[0];
		result[1] = pcallResult[1];
		if (result[0] === true) break;
		RunService.Stepped.Wait();
	}
	return result;
}
